envoy/Makefile

```diff
@@ -70,11 +102,7 @@ $(CHECK_FORMAT): force-non-root
 
 install: force-root
 	$(INSTALL) -m 0755 -d $(DESTDIR)$(BINDIR)
-	$(INSTALL) -m 0755 -T $(ENVOY_BIN) $(DESTDIR)$(BINDIR)/cilium-envoy
-# Strip only non-debug builds
-ifeq "$(findstring -dbg,$(realpath bazel-bin))" ""
-	$(STRIP) $(DESTDIR)$(BINDIR)/cilium-envoy
-endif
+	$(INSTALL) -m 0755 -T $(CILIUM_ENVOY_RELEASE_BIN) $(DESTDIR)$(BINDIR)/cilium-envoy
```

Comment:
IMO we should keep the building with specific options and stripping as two separate steps, e.g., to allow unstripped release binaries to be run in the CI.

Reply:
OK, how about having other targets which produce non-stripped binaries? I would still like to keep stripping by default.

Reply:
Or better, explicitly don't strip for a few specific targets, e.g. `envoy-debug`.

Reply:
The problem we have here it none of this matters if there is no problem. For the time being it would be really valuable to get good backtraces for any possible (if unlikely) segfaults, null dereferences etc. For that we need debug symbols and stack frame pointers. Running a non-optimized debug version would be even better, but it may be too slow. I have not experimented with that in the CI yet, one barrier being that a special box image with cached debug build would be needed. Also, when I did it locally, the unstripped binary was whopping 760MB :-)

Reply:
I guess only stripping the release builds, and building that by default only when PKG_BUILD is passed in is a reasonable compromise. That way `make` will get you a non-release build (fastbuild) that still has the debug symbols. We should then make sure that at least the runtime tests in CI are run with that.